### DISCUSSION TITLE
关于Google翻译和语音朗读功能无法正常使用的说明

### DISCUSSION THREAD

**Discussion by YiiGuxing** [YiiGuxing/TranslationPlugin#2315] [Announcements（公告）]
<h2 align="center">关于Google翻译和语音朗读无法正常使用的说明</h2>

2022年10月1日，Google突然停止了Google翻译在中国大陆的业务，不再向中国大陆区域提供翻译服务，官方给出的理由是“因为使用率低”。这一变化直接不可避免地影响到了插件内置的Google翻译引擎和基于Google翻译的语音朗读（TTS）功能，导致其无法正常使用。因此建议大家换用其他翻译引擎，微软翻译、有道翻译、百度翻译和阿里翻译都是很不错的选择。TTS 引擎也集成了微软 Edge TTS 和 OpenAI TTS。未来开发者会带来更多新的翻译引擎和语音合成（TTS）引擎供大家选择，敬请期待！

**Comment by YiiGuxing** [comment]
想要在插件中继续使用Google翻译引擎和默认语音朗读（TTS）功能的话，您可能需要以下之一：
1. 一个梯子，详情请另参阅 #2051、#2296；
2. 【2023.6.2更新】一个您当前网络环境可以访问的谷歌翻译的镜像服务器，并在谷歌翻译设置中配置（需要插件版本为3.5.0及以上）；
3. 一个您当前网络环境可以访问的世界各地（除中国大陆）的谷歌云服务器IP，然后在您的系统 `Hosts` 中添加此IP与 `translate.googleapis.com` 域名的映射:

    ```
    谷歌云服务器IP translate.googleapis.com
    ```
    > 使用此工具可以找到可用的IP：[GoogleTranslate_IPFinder](https://github.com/GoodCoder666/GoogleTranslate_IPFinder)

    > 如果您使用的是 Windows 系统，推荐您使用 [PowerToys](https://github.com/microsoft/PowerToys) 的主机文件编辑器编辑 Hosts 文件，方便又快捷。

**Reply by Harmless-Hide** [reply]
谢谢，可以了

**Reply by zxcsjf** [reply]
目前 172.253.124.90 这个可以用
谢谢！

**Reply by bububear** [reply]
可以了，谢谢，不然总感觉少点什么。

**Reply by 956825718** [reply]
可以了，栓Q

**Reply by YiiGuxing** [reply]
> 目前 172.253.124.90 这个可以用
> 谢谢！

其他的不能用了吗？

**Reply by YiiGuxing** [reply]
> 目前 172.253.124.90 这个可以用 谢谢！

@zxcsjf 我抽查了几个，可以用的嘛，别乱吓我啊。。。

**Reply by YiiGuxing** [reply]
> 现在有一些开源的本地翻译库，可以考虑提供一个选项使用这些库，对一些简单的翻译迅速有效

计划在v4.0版本推出自定义翻译引擎，那时候就可以实现这个功能了。

**Reply by 956825718** [reply]
> 想要在插件中继续使用Google翻译引擎和默认语音朗读（TTS）功能的话，您可能需要：
> 
> 1. 一个梯子，详情请另参阅 [谷歌翻译无法访问 #2051](https://github.com/YiiGuxing/TranslationPlugin/issues/2051)、[谷歌翻译文档失败: Not Found #2296](https://github.com/YiiGuxing/TranslationPlugin/issues/2296)；
> 2. 或者一个您当前网络环境可以访问的世界各地（除中国大陆）的谷歌云服务器IP，然后在您的系统 `Hosts` 中添加此IP与 `translate.googleapis.com` 域名的映射:
> 
> ```
> 谷歌云服务器IP translate.googleapis.com
> ```
> 
> 可用的谷歌云服务器IP：
> 
> ```
> # 2022.10.25 更新
> # 如果您发现此IP列表中有无法使用的或者发现了新可用的，请告诉我，我会定期更新。
> 
> 142.250.4.90
> 172.253.114.90
> 172.217.203.90
> 172.253.112.90
> 142.250.9.90
> 172.253.116.90
> 142.250.97.90
> 142.250.30.90
> 142.250.111.90
> 172.217.215.90
> 142.250.11.90
> 142.251.9.90
> 108.177.122.90
> 142.250.96.90
> 142.250.100.90
> 142.250.110.90
> 172.217.214.90
> 172.217.222.90
> 142.250.31.90
> 142.250.126.90
> 142.250.10.90
> 172.217.195.90
> 172.253.115.90
> 142.251.5.90
> 142.250.136.90
> 142.250.12.90
> 142.250.101.90
> 172.217.192.90
> 142.250.0.90
> 142.250.107.90
> 172.217.204.90
> 142.250.28.90
> 142.250.125.90
> 172.253.124.90
> 142.250.8.90
> 142.250.128.90
> 142.250.112.90
> 142.250.27.90
> 142.250.105.90
> 172.253.126.90
> ```
> 
> > 如果您使用的是 Windows 系统，推荐您使用 [PowerToys](https://github.com/microsoft/PowerToys) 的主机文件编辑器编辑 Hosts 文件，方便又快捷。

这一批好像都失效的差不多了

**Reply by YiiGuxing** [reply]
@956825718 感谢提醒，我重新测试并更新了列表，确实差不多都失效了，就只剩下5个可用的。。。。

**Reply by 956825718** [reply]
不客气@YiiGuxing

**Reply by xkongbuwuyu** [reply]
#可以试试这批host

108.177.111.90
108.177.122.90
108.177.126.90
108.177.127.90
108.177.97.100
142.250.1.90
142.250.10.90
142.250.100.90
142.250.102.90
142.250.103.90
142.250.105.90
142.250.107.90
142.250.11.90
142.250.111.90
142.250.112.90
142.250.113.90
142.250.114.90
142.250.115.90
142.250.12.90
142.250.123.90
142.250.125.90
142.250.126.90
142.250.128.90
142.250.13.90
142.250.136.90
142.250.138.90
142.250.141.90
142.250.142.90
142.250.145.90
142.250.147.90
142.250.148.90
142.250.149.90
142.250.152.90
142.250.153.90
142.250.157.90
142.250.158.90
142.250.159.90
142.250.27.90
142.250.28.90
142.250.30.90
142.250.31.90
142.250.8.90
142.250.9.90
142.250.96.90
142.250.97.90
142.250.98.90
142.250.99.90
142.251.0.90
142.251.1.90
142.251.10.90
142.251.107.90
142.251.111.90
142.251.112.90
142.251.116.90
142.251.117.90
142.251.12.90
142.251.120.90
142.251.15.90
142.251.16.90
142.251.160.90
142.251.161.90
142.251.162.90
142.251.163.90
142.251.164.90
142.251.165.90
142.251.166.90
142.251.167.90
142.251.169.90
142.251.18.90
142.251.2.90
142.251.31.90
142.251.4.90
142.251.5.90
142.251.6.90
142.251.8.90
142.251.9.90

**Reply by Matthew2Qiao** [reply]
到2023年8月28日，还有两个有用
108.177.97.100（比较稳定）
142.250.8.90（有概率超时）

**Comment by malzahar-mz** [comment]
有道那个语音合成小姐姐声音很好听的，期待能添加进来。。。💕

**Comment by ljc001** [comment]
感谢大佬，终于又可以正常使用了

**Comment by tomtrigger** [comment]
very good

**Comment by syrm-ll** [comment]
百度翻译后会丢失排版
阿里的翻译字数限制太低了... 不到10行的 javadoc 都不给翻译 😒  有可能阿里的翻译按照字数限制截取文本逐句翻译吗

**Reply by YiiGuxing** [reply]
阿里翻译一次翻译字数限制有5000字符呢：
![image](https://user-images.githubusercontent.com/10737066/202738574-63cfcd94-d79e-4a54-bd5e-d072e38cc4b0.png)
这个数字也不低了，超过的可以手动选择句子翻译嘛。

**Reply by syrm-ll** [reply]
![image](https://user-images.githubusercontent.com/40515646/202848185-f2b2cbba-91dd-4739-809b-ccd508447c35.png)

是我哪里设置的不对吗?  同样是 `java.lang.String#split(java.lang.String, int)`  的注释

**Reply by YiiGuxing** [reply]
这就奇怪了，你用的是哪个版本的插件？

**Reply by syrm-ll** [reply]
插件 3.3.5
IDEA 2022.2.3 (222.4345.14)
阿里云用的子账号 只有 `AliyunMTReadOnlyAccess` `AliyunMTFullAccess` 两个权限

**Reply by YiiGuxing** [reply]
那请你提供一下IDE的日志，我拿来分析分析。

**Reply by syrm-ll** [reply]
我一般在阅读器模式下右键翻译文档, 有时候用悬浮的 doc 窗口自动翻译  都是这样, 感觉大概1000 或者 2000 字就会触发这种限制

我在一个干净的演示项目里尝试了几次成功的翻译, 最后有几次尝试翻译 `java.lang.String` 类注释不成功, 附日志: (我没发现什么有用的日志)

[idea.log](https://github.com/YiiGuxing/TranslationPlugin/files/10056409/idea.log)

**Reply by syrm-ll** [reply]
刚才试着取消阅读器模式, 自己选中注释发现可以翻译 😂 但是丢失了所有排版和链接效果
从日志来看 `java.lang.String` 类注释大概 3000+字
![image](https://user-images.githubusercontent.com/40515646/203054639-2aadc102-58a0-4625-a2f3-f496418957e1.png)

**Reply by YiiGuxing** [reply]
你这个问题太奇怪了，看来只能通过debug才能找到问题了，但我这边又不能重现你的问题。。。
我怀疑你IDE生成的文档带得有一些奇怪的字符，像路径，多余的HTML标签、属性之类的，这些会被计数但又不会渲染出来。

**Comment by StarLeet** [comment]
现在谷歌翻译不论是挂t设置代理还是改hosts都翻译失败了……有人有这样情况吗？

**Reply by YiiGuxing** [reply]
大概是你的网络环境的问题了，我使用的是 142.250.4.90 这个IP，一直能用的。

**Reply by StarLeet** [reply]
> 大概是你的网络环境的问题了，我使用的是 142.250.4.90 这个IP，一直能用的。

我电脑是可以直接打开谷歌翻译com网站的，之前也确实可以直接翻译。但是前几天突然不行了。我还以为中间更新过插件，但是其实没有。我晚点再试试看。

**Reply by StarLeet** [reply]
> 大概是你的网络环境的问题了，我使用的是 142.250.4.90 这个IP，一直能用的。

ok我排出问题在哪了……我前几天，给hosts文件多追加几个ip的时候，在com后面多加了个s。我之前检查了几次都没发现。刚才发现不对劲，应该是我上次弄的时候ctrl c，擦到s了。现在正常了。

**Comment by 1052915220** [comment]
可惜了语音朗读

**Reply by YiiGuxing** [reply]
语音朗读可以使用上面提的修改HOSTS的方法继续使用，后面会添加其他的语音朗读引擎的。

**Reply by LHdongMars** [reply]
我也是不能朗读，我还以为是我电脑代理的问题。

**Comment by RainySecret** [comment]
大佬666，集成了微软翻译和DeepL翻译💕

**Reply by onlybigroc** [reply]
deepl账号怎么订阅呢，我试了国内的信用卡，但失败了

**Reply by selfdialog** [reply]
万能的淘宝

**Comment by hanbing5210** [comment]
对于  google   我只想说一句话   你放屁

**Comment by YiiGuxing** [comment]
谷歌翻译目前在3.4.0版本已经有替代方案了，下一步咱们的目标是TTS的代替方案。💪💪💪

**Reply by youyuanme** [reply]
什么替代方案，目前我用的3.4.2版本为啥没有了字节跳动的翻译引擎选项了呢？我记得有个版本是有的 💪💪💪

**Reply by YiiGuxing** [reply]
？？？目前并没有接入字节跳动翻译引擎呢，是接入了微软翻译引擎，可以直接使用，不需要魔法。

**Reply by youyuanme** [reply]
不是有一个火山翻译吗？我忘了是哪个版本有了，换了电脑之后重新装了插件就没有了.....

**Reply by YiiGuxing** [reply]
呃...那你应该使用的不是我的这个插件，我的插件还没有接入火山翻译。

**Reply by youyuanme** [reply]
哦哦估计是我记错了，不过可以考考一下接入火山翻译。阿里翻译引擎好像没有TTS？

**Reply by YiiGuxing** [reply]
后面再考虑，目前任务积累太多没有处理了。。。

**Comment by onlybigroc** [comment]
deepl账号怎么订阅呢，我试了国内的信用卡，但失败了

**Reply by YiiGuxing** [reply]
deepl账号不支持开户地址在国内的信用卡，目前只支持国外的。

**Comment by MaiJiantian** [comment]
这个亲测可以用

[Google翻译修复工具_1.3.exe.zip](https://github.com/YiiGuxing/TranslationPlugin/files/10137601/Google._1.3.exe.zip)

v1.3
修改可用IP
新增接口可用性检测，能在设置IP之前，先检测此IP是否可用，以应对有些IP虽然能联通，但是不能使用
新增清除host菜单项目，可用快速删除设置的Google翻译HOST一项，不影响其他的HOST。

v1.2
修复1.1版本设置IP无效问题
新增单条IP延迟测试

v1.1
修改HOST替换方式，不再覆盖原来HOST的内容
新增菜单打开HOST，可以一键打开HOST文件，查看内容
IP列表新增右键菜单，可以选择指定IP进行设置

**Comment by qihao3768** [comment]


**Reply by YiiGuxing** [reply]
???

**Comment by l1n6yun** [comment]
希望能修复方法翻译的排版
![图片](https://user-images.githubusercontent.com/19780706/210034946-554d8686-30b2-41cc-a8ad-ce6504b3d624.png)

**Reply by YiiGuxing** [reply]
对于有道翻译和百度翻译，文档的翻译是没有排版的。想要有排版的文档，只能使用其他的翻译引擎。

**Comment by lesliechueng** [comment]
梯子地址https://dler.pro/auth/register?affid=144153

**Reply by RainySecret** [reply]
请问这个梯子地址注册了后有什么用？

**Reply by lesliechueng** [reply]
正常访问谷歌那些被墙掉的网站啊。。。。

**Comment by wanghaora** [comment]
2023年2月6日
临时解决朗读问题：
现在除了vpn外还有个改host的办法，具体详见[https://github.com/hcfyapp/google-translate-cn-ip](https://github.com/hcfyapp/google-translate-cn-ip) 这里包含能用的ip
但是用之前需要先测试一下可以在[https://github.com/Ponderfly/GoogleTranslateIpCheck](https://github.com/Ponderfly/GoogleTranslateIpCheck)（图形化窗口）
[https://github.com/GoodCoder666/GoogleTranslate_IPFinder](url)（py代码）
测试之后用这个网址试一下：[https://translate.googleapis.com/translate_a/element.js](https://translate.googleapis.com/translate_a/element.js)

ipv6的host在[https://github.com/hcfyapp/crx-selection-translate/discussions/1607#discussioncomment-4654678](https://github.com/hcfyapp/crx-selection-translate/discussions/1607#discussioncomment-4654678)的评论中会有一些


改镜像：下面这两个镜像地址似乎都会在晚上大概 8 点后失效，然后在第二天早上恢复
https://translate.3822808.com
https://translate.amz.wang

2023年2月6日

**Reply by wanghaora** [reply]
142.250.105.90 translate.google.com
142.250.105.90 translate.googleapis.com
目前我用的

**Reply by 956825718** [reply]
> 142.250.105.90 translate.google.com 142.250.105.90 translate.googleapis.com 目前我用的
还可以用吗

**Reply by wanghaora** [reply]
> > 142.250.105.90 translate.google.com 142.250.105.90 translate.googleapis.com 目前我用的
> 还可以用吗
> 

凉了不行了

**Comment by qigeqigeqi** [comment]
推荐个老牌机场给大家，还算稳定 https://suying00.com/auth/register?code=1uWy

**Reply by yang-chun** [reply]
用的微软也不行

**Comment by yang-chun** [comment]
> ## 关于Google翻译和语音朗读无法正常使用的说明
> 2022年10月1日，Google突然停止了Google翻译在中国大陆的业务，不再向中国大陆区域提供翻译服务，官方给出的理由是“因为使用率低”。这一变化直接不可避免地影响到了插件内置的Google翻译引擎和基于Google翻译的语音朗读（TTS）功能，导致其无法正常使用。因此建议大家换用其他翻译引擎，微软翻译、有道翻译、百度翻译和阿里翻译都是很不错的选择。未来开发者会带来更多新的翻译引擎和语音合成（TTS）引擎供大家选择，敬请期待！
用的微软也不行

**Reply by wanghaora** [reply]
估计都不行了 host凉了

**Reply by YiiGuxing** [reply]
现在TTS只有谷歌的，访问不了谷歌的网络就用不了。目前正在计划接入其他的TTS，预计在v3.5.0推出。

**Reply by wanghaora** [reply]
> 现在TTS只有谷歌的，访问不了谷歌的网络就用不了。目前正在计划接入其他的TTS，预计在v3.5.0推出。

我落下了没技术的泪水  ᶘ ᵒᴥᵒᶅ

**Reply by YiiGuxing** [reply]
@yang-chun @wanghaora host 列表重新更新了，还是有可以用的。。。

**Reply by yang-chun** [reply]
现在有可用的host吗

**Reply by wanghaora** [reply]
我没找着。。。。

**Reply by YiiGuxing** [reply]
这个好像还可以用：`172.217.192.90`

**Reply by YiiGuxing** [reply]
IP 列表已更新

**Reply by wanghaora** [reply]
上微软翻译吧5555

**Reply by yang-chun** [reply]
微软翻译什么时候能商tts

**Reply by wanghaora** [reply]
附议

**Reply by YiiGuxing** [reply]
可能在 3.5.0 或者 3.6.0，具体时间无法承诺。。。

**Reply by yang-chun** [reply]
现在有最新的ip吗

**Comment by xia930425** [comment]
很奇怪，我挂了代理，浏览器已经可以正常使用谷歌翻译了，然而idea插件还是不可以

**Reply by YiiGuxing** [reply]
那说明IDEA没有走代理，需要再配置，可参考 #2051 中相关的帖子。

**Comment by javaNewHard** [comment]
改了代理能使用了 请问可以切换 英音/美音 朗读吗

**Reply by YiiGuxing** [reply]
目前不支持。

**Comment by itcsman** [comment]
idea里原本设置的阿里翻译，但是老是会变回去谷歌翻译 怎么回事 改成阿里不久又会变回去

**Reply by YiiGuxing** [reply]
可能是你设置之后没有正常退出ide，设置在退出ide时保存。

**Comment by XiaJack** [comment]
脖子被掐了，好难受

**Comment by JimmyLi9201** [comment]
国内大厂没有自动翻译引擎吗？

**Reply by wanyi96** [reply]
都有，但是要钱，他们提供的是api接口请求翻译的那种，但是首先要用开发者资格或者公司名义申请，申请后仍然是要钱的，个人如果愿意折腾，只为了自己使用的话，也可以考虑用这种方法。毕竟相比之下，一个人使用的次数还是不贵的

**Comment by season886** [comment]
我用了有道翻译，但是语音TTS，显示连接超时，用的难道还是谷歌的TTS？

**Reply by YiiGuxing** [reply]
TTS 目前只有谷歌的，其他 TTS 还在开发中。

**Comment by chaceadi666** [comment]
能不能别整那么多弹窗，不学好的，只会学国内大厂是吧

**Reply by YiiGuxing** [reply]
那你看这个弹窗下面是不是有个“不再显示”的按钮？你点一下不就完事了嘛？
我不弹的话，不明的人可能就会说“哎呀你这根本就不能嘛，不能用那你放那干嘛？”然后反手就给我一个差评，我哪哭去？

**Reply by xiangdaojizhi** [reply]
谷歌服务器地址要填什么，从哪里找去，我有梯子不知道怎么搞

**Comment by 956825718** [comment]
> 到2023年8月28日，还有两个有用
> 108.177.97.100（比较稳定）
> 142.250.8.90（有概率超时）

好用

**Comment by MrGlp** [comment]
mark

**Comment by hanbing5210** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by MrGlp** [comment]
mark
